draft-pauly-add-requirements.md

```diff
@@ -0,0 +1,252 @@
+  -
+    ins: C. A. Wood
+    name: Christopher A. Wood
+    org: Cloudflare
+    street: 101 Townsend St
+    city: San Francisco
+    country: United States of America
+    email: caw@heapingbits.net
+  -
+    ins: P. McManus
+    name: Patrick McManus
+    org: Fastly
+    email: mcmanus@ducksong.com
+  -
+    ins: T. Jensen
+    name: Tommy Jensen
+    org: Microsoft
+    email: tojens@microsoft.com
+
+--- abstract
+
+This document describes several use cases for discovering DNS resolvers that support
+encrypted transports, and discusses how solutions for these use cases can be designed
+to use common mechanisms. It also considers the requirements for privacy and security
+when designing resolver discovery mechanisms.
+
+--- middle
+
+# Introduction
+
+Several protocols for protecting DNS traffic with encrypted transports have been defined,
+such as DNS-over-TLS (DoT) {{?RFC7858}} and DNS-over-HTTPS (DoH) {{?RFC8484}}.
+Encrypted DNS can provide many security and privacy benefits for network clients.
+
+While it is possible for clients to hard-code encrypted DNS resolvers to use, dynamic
+discovery and provisioning of encrypted resolvers can expand the usefulness and
+applicability of encrypted DNS to many more use cases.
+
+This document first describes several use cases for discovering DNS resolvers that support
+encrypted transports ({{use-cases}}).
+
+Next, it discusses how solutions for these use cases can be grouped and categorized to point
+to the usefulness of common mechanisms ({{mechanisms}}).
+
+Last, it considers the requirements for privacy and security when designing resolver discovery
+mechanisms ({{priv-sec}}).
+
+This document is designed to aid in discussion of the Adaptive DNS Discovery (ADD) working
+group as defines mechanism requirements.
+
+## Specification of Requirements
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+"OPTIONAL" in this document are to be interpreted as described in BCP 14
+{{?RFC2119}} {{?RFC8174}} when, and only when,
+they appear in all capitals, as shown here.
+
+# Use Cases {#use-cases}
+
+This section describes various use cases for which it is possible to discover an encrypted
+resolver. For each use case, the privacy and security benefits of adding encrypted resolution
+are briefly described.
+
+## Network-provisioned resolvers {#local-network}
+
+DNS servers are often provisioned by a network as part of DHCP options {{?RFC2132}} or
+IPv6 Router Advertisement (RA) options {{?RFC8106}}. These options describe one or more 
+DNS resolver IP addresses, to be used for traditional unencrypted DNS.
+
+Discovering an encrypted resolver that is provisioned by the network can provide several
+benefits:
+
+- Prevent other devices on the network from observing client DNS messages
+- Verify that answers come from the selected DNS resolver
+- Authenticate that the DNS resolver is the one provisioned by the network
```

Comment:
The authentication is a property of using an encrypted resolver, that is new as compared to an unencrypted resolver. Having any authentication of the resolver is new.

Reply:
I think this ties to the definitions below. Discovery doesn't mean authentication, at least in my usage of it. I might learn of the IP address for a resolver, but that resolver might not have an IP address cert to authenticate with. Does that mean it's been "discovered"?

Reply:
I totally agree that discovery doesn't equal authentication. This list isn't about discovery. This list is about what you get by using encrypted DNS to talk to your local resolver instead of unencrypted DNS.

Reply:
I guess I'm still confused then. The text says, "discovering an encrypted resolver that is provisioned by the network can authenticate that the DNS resolver is the one provisioned by the network," which to me says discovery of the resolver => authentication of said resolver. What am I missing?

Reply:
Okay, changed to

"Using an encrypted resolver that is provisioned by the network can provide several
benefits that are not possible if only unencrypted DNS is used:"

Reply:
👍 Much better -- thanks!